### PR TITLE
Docs collection

### PR DESCRIPTION
**Pull request by jacoblogan**
#### Description of changes
Updates to the collection primitive docs. Use the new example format, update existing examples, add grid and styling examples.

collection

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/pages/[platform]/components/collection/examples/GridCollectionExample.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      templateColumns="1fr 1fr 1fr"
+      templateRows="10rem 10rem 10rem"
+    >
+      {(item, index) => {
+        return (
+          <Card
+            key={index}
+            padding={tokens.space.medium}
+            maxWidth="180px"
+            fontSize="0.5em"
```
Comment (hbuchel) [inline docs/src/pages/[platform]/components/collection/examples/GridCollectionExample.tsx:42]: Some comment for using font size token.

File: docs/src/pages/[platform]/components/collection/react.mdx [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
+  DefaultCollectionExample,
+  GridCollectionExample,
+  ListCollectionExample,
+  LocalCollectionClassExample,
+  PaginationCollectionExample,
+  SearchCollectionExample,
+  StylingCollectionExample,
+} from './examples';
 
 A Collection wraps Flex and Grid components, and provides a way to display items in a collection from a data source.
```
Comment (hbuchel) [inline docs/src/pages/[platform]/components/collection/react.mdx:17]: Can we move this to index.page.mdx?

File: docs/src/pages/[platform]/components/collection/react.mdx [hunk 3]
```diff
@@ -20,207 +22,176 @@ A Collection wraps Flex and Grid components, and provides a way to display items
...
+```tsx file=./examples/ListCollectionExample.tsx
+
 ```
 
+  </ExampleCode>
+</Example>
+
+### Grid
+
+The `grid` collection type can be customized with and of the following Grid props: `templateColumns` and `templateRows`. Then the Collection children can use the Grid props to control their size and placement such as `row` and `column`.
```
Comment (joebuono) [inline docs/src/pages/[platform]/components/collection/react.mdx:60]: typo: "can be customized with the following Grid props"

File: docs/src/pages/[platform]/components/collection/react.mdx [hunk 4]
```diff
@@ -20,207 +22,176 @@ A Collection wraps Flex and Grid components, and provides a way to display items
...
+```tsx file=./examples/GridCollectionExample.tsx
+
+```
+
+  </ExampleCode>
 </Example>
 
 ## Pagination
 
 A Collection can be paginated, adding a special `isPaginated` property. Change the page size passing a `itemsPerPage` property (default: 10).
```
Comment (joebuono) [inline docs/src/pages/[platform]/components/collection/react.mdx:75]: nit: "A Collection can be paginated by adding a special `isPaginated` property. Change the page size by passing a `itemsPerPage` property (default: 10).

**Review by joebuono — APPROVED** [review]
Looks good, pending Heather's comments and my nits

**Review by hbuchel — APPROVED** [review]
Thanks for the updates!

**Pull request merged**